### ISSUE TITLE
Edge case unit tests for 2D Lie groups

### ISSUE THREAD

**Issue by mhmukadam** [facebookresearch/theseus#86]
If I understand correctly, this is to test different corner cases (e.g., near zero tangent vectors). If so, we should also add similar tests for SE2 and SO2 (in a new PR). But agree that adding some small comments here would be useful.

_Originally posted by @luisenp in https://github.com/facebookresearch/theseus/pull/71#discussion_r803052471_

**Comment by fantaosha** [comment]
Yes, my major concerns are the `log_maps` around `pi`. Near `zero` should not be a problem.
